network/p2p/p2p_pubsub.go

```diff
@@ -226,16 +234,26 @@ func (n *p2pNetwork) handlePubsubMessages(logger *zap.Logger) func(ctx context.C
 		if msg == nil {
 			return nil
 		}
+		// TODO: check the logic
 
-		var decodedMsg *queue.DecodedSSVMessage
-		if msg.ValidatorData != nil {
-			m, ok := msg.ValidatorData.(*queue.DecodedSSVMessage)
-			if ok {
-				decodedMsg = m
-			}
+		// var decodedMsg *queue.DecodedSSVMessage
+		// if msg.ValidatorData != nil {
+		// 	m, ok := msg.ValidatorData.(*queue.DecodedSSVMessage)
+		// 	if ok {
+		// 		decodedMsg = m
+		// 	}
+		// }
+		// if decodedMsg == nil {
+		// 	return errors.New("message was not decoded")
+		// }
+
+		signedSSVMessage := &spectypes.SignedSSVMessage{}
+		if err := signedSSVMessage.Decode(msg.GetData()); err != nil {
+			return fmt.Errorf("signed message was not decoded: %w", err)
 		}
-		if decodedMsg == nil {
-			return errors.New("message was not decoded")
+		decodedMsg, err := queue.DecodeSignedSSVMessage(signedSSVMessage)
+		if err != nil {
+			return fmt.Errorf("message was not decoded: %w", err)
```

Comment:
@nkryuchkov can you chime in on this? are we not using `ValidatorData` anymore? 
@pavelkrolevets , based on @nkryuchkov 's answer -
if `ValidatorData` not used remove the comment.
if its still being used then take the decoded message from there to not decode twice.

Reply:
`ValidatorData` is used to store a decoded message and to avoid decoding the message again

Reply:
@nkryuchkov , but as you see in the code above, the usage of ValidatorData was removed. is GetData enough here?

Reply:
@y0sher Well, it's enough to work because `ValidatorData` is an optimization that stores a message decoded from `GetData()`: https://github.com/ssvlabs/ssv/blob/alan/no-fork/message/validation/validation.go#L105

However, I don't understand why we should remove `ValidatorData` and avoid using the optimization

@pavelkrolevets Could you please clarify this?

Reply:
@nkryuchkov understood now, thank you for clarifying. Ill bring it back.